saleor/plugins/webhook/tasks.py

```diff
@@ -219,60 +222,89 @@ def send_webhook_using_http(
+        client = boto3.client(
+            "sqs",
+            region_name=region,
+            aws_access_key_id=parts.username,
+            aws_secret_access_key=(
+                unquote(parts.password) if parts.password else parts.password
+            ),
+        )
+        queue_url = urlunparse(
+            (
+                "https",
+                parts.hostname,
+                parts.path,
+                parts.params,
+                parts.query,
+                parts.fragment,
+            )
+        )
+        is_fifo = parts.path.endswith(".fifo")
+
+        msg_attributes = {
+            "SaleorDomain": {"DataType": "String", "StringValue": domain},
+            "EventType": {"DataType": "String", "StringValue": event_type},
+        }
+        if signature:
+            msg_attributes["Signature"] = {
+                "DataType": "String",
+                "StringValue": signature,
+            }
+
+        message_kwargs = {
+            "QueueUrl": queue_url,
+            "MessageAttributes": msg_attributes,
+            "MessageBody": message.decode("utf-8"),
+        }
+        if is_fifo:
+            message_kwargs["MessageGroupId"] = domain
+        with catch_duration_time() as duration:
+            response = client.send_message(**message_kwargs)
+            return WebhookResponse(content=response, duration=duration())
+    except (ClientError,) as e:
+        return WebhookResponse(content=str(e), status=EventDeliveryStatus.FAILED)
 
 
 def send_webhook_using_google_cloud_pubsub(
     target_url, message, domain, signature, event_type
 ):
-    parts = urlparse(target_url)
-    client = pubsub_v1.PublisherClient()
-    topic_name = parts.path[1:]  # drop the leading slash
-    with catch_duration_time() as duration:
-        future = client.publish(
-            topic_name,
-            message,
-            saleorDomain=domain,
-            eventType=event_type,
-            signature=signature,
-        )
-        response_duration = duration()
-        response = future.result()
-        return WebhookResponse(content=response, duration=response_duration)
+    try:
+        parts = urlparse(target_url)
+        client = pubsub_v1.PublisherClient()
+        topic_name = parts.path[1:]  # drop the leading slash
+        with catch_duration_time() as duration:
+            future = client.publish(
+                topic_name,
+                message,
+                saleorDomain=domain,
+                eventType=event_type,
+                signature=signature,
+            )
+            response_duration = duration()
+            response = future.result()
+            return WebhookResponse(content=response, duration=response_duration)
+    except (pubsub_v1.publisher.exceptions.MessageTooLargeError, RuntimeError) as e:
+        return WebhookResponse(content=str(e), status=EventDeliveryStatus.FAILED)
+
+
+def generate_request_headers(event_type, domain, signature):
```

Comment:
Do we need this function? It's used in one place and doesn't do anything except returning a dict.

Reply:
Right, I made this function before simplifying when there were more uses... Right now it is used only in one place and in tests. Maybe I'll move it to tests utils to be used and remove it from code